src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -2102,6 +2102,14 @@ private List<CompletionResult> GetResultForIdentifier(CompletionContext completi
             result = CompletionCompleters.CompleteCommandArgument(completionContext);
             replacementIndex = completionContext.ReplacementIndex;
             replacementLength = completionContext.ReplacementLength;
+
+            if (result.Count == 0
+                && completionContext.TokenAtCursor.TokenFlags.HasFlag(TokenFlags.TypeName)
+                && lastAst?.Find(a => a is MemberExpressionAst, searchNestedScriptBlocks: false) is not null)
+            {
+                result = CompletionCompleters.CompleteType(completionContext.TokenAtCursor.Text).ToList();
+            }
+
```

Comment:
I don't see new tests for the code. What is a scenario we address here?

Reply:
I think this was something like `[array]::empty[str`<tab> should complete to types known for `string`.

Reply:
You can move this to follow PR to reduce that PR.

Reply:
I don't think it makes that much sense to separate this out tbh.

Reply:
You know our experience :-) - the more compact change and the less risky the faster we can review and merge.
Since you do not prepare tests for the scenario I believe we can address the scenario in follow PR and fast merge since there is small risk.